### ISSUE TITLE
Where to put manifest.json and sw.js with Tomcat

### ISSUE THREAD

**Issue by Jantjevlith** [vincentmorneau/apex-pwa#11]
Not really an expert here so be easy on me.

![screen shot 2019-01-04 at 14 38 01](https://user-images.githubusercontent.com/33228315/50690669-78a30380-102e-11e9-8710-3c8ef3a90cb7.png)

I'm following your guide but i'm not able to find the manifest.json.

We are using Tomcat (Nginx as reverse proxy, oracle linux 7.5) , and tried multiple locations such as:

/var/lib/tomcat/webapps
/var/lib/tomcat/webapps/ords

furthermore for now we are using http instead of https (may that be the issue?).

I have tried different ways but I 'm out of options, if u could point me in a new (correct) direction it would be much appreciated.

(Also tried a it on a different server with ORDS standalone, same error (also HTTP....))

Tnx in advance,

Jan

**Comment by vincentmorneau** [comment]
HTTPS will be required for your service worker.

For your root location, do you see your APEX image prefix, like `/var/lib/tomcat/webapps/i/` or something similar?

**Comment by Jantjevlith** [comment]
yes it exists 


<img width="403" alt="screen shot 2019-01-04 at 19 48 45" src="https://user-images.githubusercontent.com/33228315/50705070-db5dc480-1059-11e9-9cd0-bc5ab0dbe0b2.png">

and i also tried putting it there but didn't succeed

**Comment by vincentmorneau** [comment]
Sorry I was confused. You need to find the docroot directory in the config file of your reverse proxy.

For nginx I'm not too familiar, but it should be similar to DocRoot=`/etc/httpd/docs`. For apache the config files are `httpd.conf` or `ssl.conf` on Linux.

**Comment by jellescheurwater** [comment]
Been working with Jan on this issue and turned out to be a SELinux policy issue in combination with specifying a location for the manifest in the Nginx configuration for our APEX installation.

1. Specifying a location for the manifest in Nginx, either by using the _root_ parameter or a location for manifest.json, turned the 404 error for the manifest.json into a 403 error.
2. Creating a custom policy to allow Nginx to read from the Tomcat socket solved the 403 error.

By doing the first, we introduced some other errors, but at least we got the manifest.json displayed as per your guide.
